packages/apollo-codegen-swift/src/language.ts

```diff
@@ -55,38 +43,263 @@ const reservedKeywords = new Set(['associatedtype', 'class', 'deinit', 'enum', '
+   */
+  static wrap(
+    start: SwiftSource,
+    maybeSource?: SwiftSource,
+    end?: SwiftSource
+  ): SwiftSource {
+    return new SwiftSource(
+      _wrap(
+        start.source,
+        maybeSource !== undefined ? maybeSource.source : undefined,
+        end !== undefined ? end.source : undefined
+      )
+    );
+  }
+
+  /**
+   * Given maybeArray, return an empty string if it is null or empty, otherwise return all items
+   * together separated by separator if provided.
+   *
+   * This is just a wrapper for `join()` from apollo-codegen-core/lib/utilities/printing.
+   *
+   * @param separator The separator to put between elements. This is typed as `string` with the
+   * expectation that it's generally something like `', '` but if it contains identifiers it should
+   * be escaped.
+   */
+  static join(
+    maybeArray?: (SwiftSource | undefined)[],
+    separator?: string
+  ): SwiftSource {
+    return new SwiftSource(_join(maybeArray, separator));
+  }
+}
+
+/**
+ * Template tag for producing a `SwiftSource` value by escaping expressions.
+ *
+ * All interpolated expressions will undergo identifier escaping unless the expression value is of
+ * type `SwiftSource`. If any interpolated expressions are actually intended as string literals, use
+ * the `SwiftSource.string()` function on the expression.
+ */
+export function swift(
+  literals: TemplateStringsArray,
+  ...placeholders: any[]
+): SwiftSource {
+  let result = literals[0];
+  placeholders.forEach((elt, i) => {
+    result += _escape(elt);
+    result += literals[i + 1];
+  });
+  return new SwiftSource(result);
+}
+
+function _escape(elt: any): string {
+  if (elt instanceof SwiftSource) {
+    return elt.source;
+  } else if (typeof elt === "string") {
+    return SwiftSource.identifier(elt).source;
+  } else if (Array.isArray(elt)) {
+    // I don't know why you'd be interpolating an array, but let's recurse into it.
+    return elt.map(_escape).join();
+  } else if (typeof elt === "object") {
+    // use `${…}` instead of toString to preserve string conversion semantics from untagged
+    // template literals.
+    return SwiftSource.identifier(`${elt}`).source;
+  } else if (elt === undefined) {
+    return "";
   } else {
-    return identifier;
+    // Other primitives don't need to be escaped.
+    return `${elt}`;
   }
 }
 
+// Convenience accessors for wrap/join
+const { wrap, join } = SwiftSource;
+
 export class SwiftGenerator<Context> extends CodeGenerator<
   Context,
-  { typeName: string }
+  { typeName: string },
+  SwiftSource
 > {
   constructor(context: Context) {
     super(context);
   }
 
   multilineString(string: string) {
-    this.printOnNewline(`"${escapedString(string)}"`);
+    // Disable trimming if the string contains """ as this means we're probably printing an
+    // operation definition where trimming is destructive.
+    this.printOnNewline(
+      SwiftSource.string(string, /* trim */ !string.includes('"""'))
```

Comment:
the inline comment here is to indicate what that parameter is, correct?

Reply:
Yep. Would you prefer I remove it, or use a different style?

Reply:
Nah, I just hadn't seen that particular syntax before